### ISSUE TITLE
Compiled dist version does not include classnames module

### ISSUE THREAD

**Issue by topgun743** [elementalui/elemental#85]
As per instructions of @nkbt did following:

```
git clone git@github.com:elementalui/elemental.git
cd elemental
npm install
npm run build
```

Took following files from dist folder:
elemental.css
elemental.js

Added these two files to my HTML page header. Executed the page but it complained following:
```
Error: Module name "classnames" has not been loaded yet for context: _. Use require([])
http://requirejs.org/docs/errors.html#notloaded
```
I guess full js output (independent of any dependency) is still not coming.

Reference issue: https://github.com/elementalui/elemental/issues/39#issuecomment-149583456

**Comment by JedWatson** [comment]
We need to include "classnames" in the Bower dependencies, as per [react-select](https://github.com/JedWatson/react-select/blob/master/bower.json)

**Comment by JedWatson** [comment]
@topgun743 sorry that's more an answer for the Bower dist, not your issue here.

If you're making your own custom build, you'll need to include dist builds of the other dependencies as well:

* `blacklist`
* `classnames`
* `react`
* `react-addons-css-transition-group`
* `rect-dom`

I don't think it's a good idea for us to generate these into our dist build, as you will probably want to manage them independently (along with other dependencies your app has)

**Comment by nkbt** [comment]
+ `react-addons-css-transition-group`
+ `react-dom`

Not published to bower unfortunately.

**Comment by JedWatson** [comment]
Just read a bunch of issues in the React repo around this, still unclear. Seems like those packages (`react-dom` and `react-addons-*` friends) _should_ either be included in the React dist on bower / cdnjs / etc, or available as their own packages.

@zpao @spicyj any chance you guys could clarify how we're supposed to depend on those for bower?

**Comment by nkbt** [comment]
For my libs I found the best way to test that build is correct, to create a Codepen demos. Sometimes it takes a bit of time, but then I am pretty sure it is not broken.

For example http://codepen.io/nkbt/pen/MarzEg?editors=101 (codepen demo for react-collapse)

**Comment by zpao** [comment]
I don't have a good answer for you. The whole Bower thing is a mess and really doesn't play well with what we're doing for our npm packages, which is to reach into the react project and export a specific module. For some of the addons they could be pulled out and built standalone, others will continue to reach into react so may only be available for the browser with prepackaged builds (react-with-addons). We'll see.

For react-dom, it's a part of the `react` Bower package, will be added to `main` in 0.14.1.

**Comment by nkbt** [comment]
@zpao for my modules I've created a separate org and wrote some scripts to automatically publish them to bower, so I don't have a mess in my main repos: https://github.com/nkbt/react-component-template

But I totally agree that the whole bower thing is a mess. Unfortunately too many people rely on it =(.

**Comment by brunowego** [comment]
+1

**Comment by magnuslidman** [comment]
So I'm trying to play around with elemental UI without using browserify - just use the correct set of precompiled js script tags. I'm not having any luck (admittedly I am new to the whole react and transpiling world), I can't for the life of me get rid of the "Cannot find module 'classnames' " error - I've installed the classname module from bower and inluded the index.js in my script tags, but that doesn't seem to change anything. Is Elemental supposed to be able to be used on it's own, or does it need to be compiled into a bundle using gulp etc before it is usable?

**Comment by nkbt** [comment]
@magnuslidman did you include `classnames` *before* `elemental`?

**Comment by magnuslidman** [comment]
Yes I did. With the risk of showing my dirty underwear to the world, this is the html i have:

	
 	<script src="https://cdnjs.cloudflare.com/ajax/libs/babel-core/5.8.23/browser.min.js"></script>
    <script src="/components/react/react.min.js"></script>
    <script src="/components/react/react-dom.min.js"></script>
    <script src="/components/jquery/dist/jquery.min.js"></script>
    <script src="/components/classnames/index.js"></script>
   	<script src="/components/elemental/elemental.js"></script>
	<link rel="stylesheet" href="/static/css/main.css">
	<link rel="stylesheet" href="/components/elemental/elemental.css">
	<link href='https://fonts.googleapis.com/css?family=Open+Sans' rel='stylesheet' type='text/css'>

-->

(I am serving up the javascrupt myself - only the babel js is from a CDN)

**Comment by nkbt** [comment]
That looks like a problem. Though, bower issue (and deps) is going to be
solved very soon with npmcdn. I'm working to address the issue asap.

**Comment by magnuslidman** [comment]
looking at the code, classnames seems to be a real simple construct - is there another js library I am missing (that would make the elemental js link the classnames export?

 Also - can I help out in any way to help you reproduce (if it truly is a bug - I can compile and run the example site fine using npm run build on the elemental source)

**Comment by nkbt** [comment]
I reckon the problem is somewhere in the build process, which produces `elemental.js` and does not handle external dependencies well.

No worries, that is quite easy to reproduce.

**Comment by nkbt** [comment]
Gotcha! 

@JedWatson note the letter case: `classnames` and `classNames`

<img width="652" alt="20151225-233823" src="https://cloud.githubusercontent.com/assets/175264/12002899/a64b74f6-ab60-11e5-8681-a87bfae453fb.png">

**Comment by nkbt** [comment]
I made a PR for classnames issue: #113 

Though there is another one. `blacklist` is not meant to be used without build step. I made a PR to fix that https://github.com/dcousens/blacklist/pull/6 too

**Comment by nkbt** [comment]
Reopen since the issue with `blacklist` is still not solved.

**Comment by nkbt** [comment]
Not only blacklist is a problem apparently. `react-addons-css-transition-group` also relies on build step and currently cannot be used in non-build environment. I will deal with it.

**Comment by nkbt** [comment]
Closed in favour of #118.
